addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/parserapi/ParserApi.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.commonlib.parserapi;
+
+import java.io.IOException;
+import java.util.List;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.parserapi.impl.JavaScriptLexer;
+import org.zaproxy.addon.commonlib.parserapi.impl.JavaScriptParser;
+
+public class ParserApi {
```

Comment:
This needs a proper API to extract/parse the JavaScript content. While this code might still be useful for more than one scan rule it needs to be cleaned up first.

Reply:
I guess we could remove the changelog entry and not bother announcing it until the other part is done? Or we can wait and expand it?I'm happy to push it further, just needed feedback. Do we need to start a doc?